### PR TITLE
Performance enhancement

### PR DESCRIPTION
**Pull request by bjansen**

### CODE REVIEW

**Comment by gavinking** [comment]
Looks good to me!

File: src/com/redhat/ceylon/model/typechecker/model/Package.java [hunk 1]
```diff
@@ -221,6 +222,8 @@ public TypeDeclaration getInheritingDeclaration(Declaration d) {
                 getModule()
                     .getAvailableDeclarations(
                             startingWith, proximity);
+        Map<String, DeclarationWithProximity> thingsToAdd = new HashMap<>();
```
Comment (gavinking) [inline src/com/redhat/ceylon/model/typechecker/model/Package.java:225]: @bjansen y'know, you could just build a `List` of the `namesToAdd`. Not sure if that would be faster or slower.

File: src/com/redhat/ceylon/model/typechecker/model/Package.java [hunk 2]
```diff
@@ -236,9 +239,10 @@ public TypeDeclaration getInheritingDeclaration(Declaration d) {
         		}
         	}
     		if (!already) {
-                result.put(name, existing);
+                thingsToAdd.put(name, existing);
     		}
```
Comment (quintesse) [inline src/com/redhat/ceylon/model/typechecker/model/Package.java:243]: Tabs being used here, never ever use tabs @bjansen ! ;)
They will have to be removed.

Comment (bjansen) [inline src/com/redhat/ceylon/model/typechecker/model/Package.java:243]: I kept indentation as it was before my changes :)

Comment (quintesse) [inline src/com/redhat/ceylon/model/typechecker/model/Package.java:243]: Indeed, sorry ;)

**Pull request merged**